### PR TITLE
#27 Create validator for create candidate

### PR DESCRIPTION
**Pull request by ProkopenkovDaniel**

### CODE REVIEW

File: Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using Recrutify.Services.DTOs;
 
 namespace Recrutify.Host.Configuration.Profiles
-{
+{
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/CandidateProfile.cs:6]: you have a lot of warnings: fix them

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿using FluentValidation;
+using Recrutify.Services.DTOs;
+
+namespace Recrutify.Services.Validators
+{
+    public abstract class BaseCandidateValidator<TDTO> : AbstractValidator<TDTO>
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:6]: you don't need base, just leave CreateCandidateValidator

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        where TDTO : CandidateCreateDTO
+    {
+        public BaseCandidateValidator()
+        {
+            RuleFor(c => c.Name)
+                .NotNull()
+                .NotEmpty()
+                .MaximumLength(128);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:14]: may be reduce to 50? it's person name, same for surnmae

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            RuleFor(c => c.Surname)
+                .NotNull()
+                .NotEmpty()
+                .MaximumLength(128);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:18]: reduce number

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                .NotNull()
+                .NotEmpty();
+            RuleFor(c => c.BestTimeToConnect)
+                .NotNull()
+                .NotEmpty();
+            RuleForEach(c => c.BestTimeToConnect)
+                .NotNull()
+                .NotEmpty();
+            RuleFor(c => c.GoingToExadel)
+                .NotNull();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:54]: remove this rule, can bool field be null?

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                .NotNull()
+                .NotEmpty();
+            RuleFor(c => c.RegistrationDate)
+                .NotNull()
+                .NotEmpty();
+            RuleFor(c => c.BestTimeToConnect)
+                .NotNull()
+                .NotEmpty();
+            RuleForEach(c => c.BestTimeToConnect)
+                .NotNull()
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:51]: remove, it's int can't be null

File: Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            RuleFor(c => c.Name)
+                .NotNull()
+                .NotEmpty()
+                .MaximumLength(128);
+            RuleFor(c => c.Surname)
+                .NotNull()
+                .NotEmpty()
+                .MaximumLength(128);
+            RuleFor(c => c.EnglishLevel)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/BaseCandidateValidator.cs:19]: remove rule

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                .MaximumLength(50);
+            RuleFor(c => c.PhoneNumber)
+                .NotNull()
+                .NotEmpty()
+                .Matches(@"^\+?(\d[\d\- ]+)?(\(?[\d\- ]+\))?([\d\- ])+\d$");
+            RuleFor(c => c.Email)
+                .NotNull()
+                .NotEmpty()
+                .EmailAddress();
+            RuleFor(c => c.Contacts)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:26]: add check that skype is required

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                .NotNull()
+                .NotEmpty();
+            RuleFor(t => t.Contacts.ToList()).
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:30]: remove ToList

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+                Must(t => t.Where(list => list.Type == "Skype").Count() > 0)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: Where can be changed to Any, then you shouldn't write Count() > 0

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs (hunk 9 above)
Comment (elizavetapavliv) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:30]: rename t to c

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs (hunk 10 above)
Comment (elizavetapavliv) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: rename list to contact

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: move "Skype" to const field of validator

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+                .Matches(@"^\+?(\d[\d\- ]+)?(\(?[\d\- ]+\))?([\d\- ])+\d$");
+            RuleFor(c => c.Email)
+                .NotNull()
+                .NotEmpty()
+                .EmailAddress();
+            RuleFor(c => c.Contacts)
+                .NotNull()
+                .NotEmpty();
+            RuleFor(t => t.Contacts.ToList()).
+                Must(t => t.Any(list => list.Type == "Skype"))
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: Rename t to c, list to contact - it's not list it's single object. Move Skype to const string field of validator class

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+                .NotEmpty()
+                .Matches(@"^\+?(\d[\d\- ]+)?(\(?[\d\- ]+\))?([\d\- ])+\d$");
+            RuleFor(c => c.Email)
+                .NotNull()
+                .NotEmpty()
+                .EmailAddress();
+            RuleFor(c => c.Contacts)
+                .NotNull()
+                .NotEmpty();
+            RuleFor(c => c.Contacts).
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:30]: move dot on next line - before Must

File: Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+                Must(c => c.Any(contact => contact.Type == "Skype"))
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: move Skype to constant field of validator class

Comment (ProkopenkovDaniel) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs:31]: String Skype?

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs]: yes

Comment (elizavetapavliv) [reply Recrutify/Recrutify.Services/Validators/CreateCandidateValidator.cs]: and move dot to next lune from 30 to 31

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**